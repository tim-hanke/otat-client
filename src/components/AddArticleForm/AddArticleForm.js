import React, { Component } from "react";
import ArticleApiService from "../../services/article-api-service";
import { Button, Input } from "../Utils/Utils";

export default class AddArticleForm extends Component {
  static defaultProps = {
    onAddSuccess: () => {},
  };

  state = { error: null, isFetching: false };

  handleSubmitUrl = (ev) => {
    ev.preventDefault();
    this.setState({ error: null, isFetching: true });
    const { url } = ev.target;

    ArticleApiService.postArticle(url.value)
      .then((_res) => {
        url.value = "";
        this.props.onAddSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error, isFetching: false });
      });
  };

  render() {
    const { error, isFetching } = this.state;
    return (
      <form className="AddArticleForm" onSubmit={this.handleSubmitUrl}>
        <div className="url">
          <label htmlFor="AddArticleForm__url">URL</label>
          <Input required name="url" id="AddArticleForm__url"></Input>
        </div>
        {error && (
          <div role="alert">
            <p className="red">{error}</p>
          </div>
        )}
        <Button type="submit">
          {isFetching ? "Getting URL" : "Submit URL"}
        </Button>
      </form>
    );
  }
}
