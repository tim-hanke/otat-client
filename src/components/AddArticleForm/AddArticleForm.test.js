import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import AddArticleForm from "./AddArticleForm";

describe.only("AddArticleForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AddArticleForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<AddArticleForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
