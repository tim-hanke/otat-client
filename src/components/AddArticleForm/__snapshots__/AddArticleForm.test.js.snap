// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddArticleForm component renders the UI as expected 1`] = `
<form
  className="AddArticleForm"
  onSubmit={[Function]}
>
  <div
    className="url"
  >
    <label
      htmlFor="AddArticleForm__url"
    >
      URL
    </label>
    <input
      className="Input "
      id="AddArticleForm__url"
      name="url"
      required={true}
    />
  </div>
  <button
    className="Button "
    type="submit"
  >
    Submit URL
  </button>
</form>
`;
